### DISCUSSION TITLE
building with conan as a non-dev user

### DISCUSSION THREAD

**Discussion by tstack** [friendlyanon/cmake-init#51] [Q&A]
There are instructions and presets for building a project with conan dependencies as a developer, but I'm not clear how a regular user would do a build.  Can someone point me in the right direction?

Right now, if I run conan to install packages:

```
$ conan install . -if conan -s build_type=Release -b missing
```

And then run cmake as mentioned in `BUILDING.md`:

```
cmake -S . -B build -D CMAKE_BUILD_TYPE=Release
```

It fails because it can't find the conan packages since the toolchain is not setup as it is when using the dev preset.  Should I be adding a "user" preset that does the relevant setup?  Is that something that should be part of `cmake-init`?

(Pardon my ignorance if I'm missing something obvious...)

**Comment by friendlyanon** [comment]
The reason why there is no more detailed or direct set of instructions in the user facing document to build projects is similar to why there are no packaging instructions at all; the sky's the limit and the projects can be built in any number of ways, which is intentional, as that is one of the goals of cmake-init.

For the developer workflow, there is at least one reasonable path to assume as the default that the developer wishes to take, so there it's easy to set things up.

There is no universal set of instructions for consumers, as one person's requirements could be wildly different from the next one's. https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html contains a huge number of variables that all are useful to someone somewhere and cmake-init makes no attempts to prevent any of those variables from just working.

---

For these reasons, I just left the consumer facing document as minimal as possible.

**Reply by tstack** [reply]
I think I'm expecting some common convention that would work across projects.  For example, when I download an autoconf-based project I know I can run:

```
$ ./configure
$ make
$ make install
```

For a cmake-based project, I would normally run:

```
$ cmake -S . -B build
$ cmake --build build
```

But, those commands don't seem to work for a conan-based project created by `cmake-init`.  It's fine if I have to run an extra conan step or something else, but having to specify the toolchain file on the command-line is too much.  I don't see why there would be a convention for dev-mode, but not one for the non-dev mode.

**Reply by friendlyanon** [reply]
I don't think Conan (or vcpkg for that matter) are applicable as end-user package managers. They are more focused on developers' experience. For end-users, there are software distributions.

However, one could build all the dependencies as static libraries and roll those into the resulting binary, but that obviously does not scale.
